svgdigitizer/svgplot.py

```diff
@@ -204,22 +220,182 @@ class SVGPlot:
+
+        EXAMPLES::
+
+            >>> from svgdigitizer.svg import SVG
+            >>> from io import StringIO
+            >>> svg = SVG(StringIO(r'''
+            ... <svg>
+            ...   <g>
+            ...     <path d="M 0 200 L 0 100" />
+            ...     <text x="0" y="200">E1: 0 cm</text>
+            ...   </g>
+            ...   <g>
+            ...     <path d="M 100 200 L 100 100" />
+            ...     <text x="100" y="200">E2: 1cm</text>
+            ...   </g>
+            ...   <g>
+            ...     <path d="M -100 100 L 0 100" />
+            ...     <text x="-100" y="100">j1: 0</text>
+            ...   </g>
+            ...   <g>
+            ...     <path d="M -100 0 L 0 0" />
+            ...     <text x="-100" y="0">j2: 1 A</text>
+            ...   </g>
+            ... </svg>'''))
+            >>> plot = SVGPlot(svg)
+            >>> plot.xlabel
+            'E'
+
+            >>> plot = SVGPlot(svg, page_orientation=PageOrientation.LANDSCAPE)
+            >>> plot.xlabel
+            'j'
+
+        """
+
+        for axis, orientation in self.axis_orientations.items():
+            if orientation == AxisOrientation.VERTICAL:
+                return axis
+
+        return "y"
+
+    @property
+    @cache
+    def axis_orientations(self):
+        r"""
+        Return the orientation for each axis.
+        The current implementation does not work for extrem cases where a plot is tilted by 45°.
+
+        EXAMPLES::
+
+            >>> from svgdigitizer.svg import SVG
+            >>> from io import StringIO
+            >>> svg = SVG(StringIO(r'''
+            ... <svg>
+            ...   <g>
+            ...     <path d="M 0 200 L 0 100" />
+            ...     <text x="0" y="200">E1: 0 cm</text>
+            ...   </g>
+            ...   <g>
+            ...     <path d="M 100 200 L 100 100" />
+            ...     <text x="100" y="200">E2: 1cm</text>
+            ...   </g>
+            ...   <g>
+            ...     <path d="M -100 100 L 0 100" />
+            ...     <text x="-100" y="100">j1: 0</text>
+            ...   </g>
+            ...   <g>
+            ...     <path d="M -100 0 L 0 0" />
+            ...     <text x="-100" y="0">j2: 1 A</text>
+            ...   </g>
+            ... </svg>'''))
+            >>> plot = SVGPlot(svg)
+            >>> plot.axis_orientations
+            {'E': <AxisOrientation.HORIZONTAL: 'horizontal'>, 'j': <AxisOrientation.VERTICAL: 'vertical'>}
+
+            >>> plot = SVGPlot(svg, page_orientation=PageOrientation.LANDSCAPE)
+            >>> plot.axis_orientations
+            {'E': <AxisOrientation.VERTICAL: 'vertical'>, 'j': <AxisOrientation.HORIZONTAL: 'horizontal'>}
+
+        """
+
+        axis_orientations = {}
+
+        for axis, points in self._grouped_ref_points.items():
+            # distinguish between 2 ref points
+            if len(points) == 2:
+                delta_x = abs(points[0].points[0][0] - points[1].points[0][0])
```

Comment:
Why `points[0].points[0]`. Shouldn't this be `points[0].far`?

Reply:
True.